### PR TITLE
Delete old ProvReqs

### PR DESCRIPTION
**Pull request by yaroslava-serdiuk**
#### What type of PR is this?

/kind feature

### CODE REVIEW

File: cluster-autoscaler/processors/provreq/processor.go [hunk 1]
```diff
@@ -106,7 +107,24 @@ func (p *provReqProcessor) Process(provReqs []*provreqwrapper.ProvisioningReques
...
 
 // Cleanup cleans up internal state.
-func (p *provReqProcessor) CleanUp() {}
+func (p *provReqProcessor) Cleanup() {}
+
+// DeleteOldProvReqs delete ProvReq that have terminal state (Provisioned/Failed == True) more than a week.
+func (p *provReqProcessor) DeleteOldProvReqs(provReqs []*provreqwrapper.ProvisioningRequest) {
+	for _, provReq := range provReqs {
+		conditions := provReq.Status.Conditions
+		provisioned := apimeta.FindStatusCondition(conditions, v1beta1.Provisioned)
```
Comment (kisieland) [inline cluster-autoscaler/processors/provreq/processor.go:120]: shouldn't we have booking expired here?
Can you add a test case for it as well, I think in the test case we only have one for old-failed

Comment (yaroslava-serdiuk) [reply cluster-autoscaler/processors/provreq/processor.go:120]: Currently that doesn't make any difference to check BookingExpired condition or Provisioned, however in general I think every ProvReq should have Provisioned condition, while BookingExpired is not necessary, so I decided to use Provisioned condition. 
Do you have a strong opinion to use BookingExpired here?

Comment (kisieland) [reply cluster-autoscaler/processors/provreq/processor.go:179]: No, just curious why this one was used

**Review by kisieland — APPROVED** [review]
/lgtm

File: cluster-autoscaler/processors/provreq/processor_test.go [hunk 2]
```diff
@@ -150,12 +151,55 @@ func TestRefresh(t *testing.T) {
...
 		pr.CreationTimestamp = metav1.NewTime(test.creationTime)
 		pr.Spec.ProvisioningClassName = v1beta1.ProvisioningClassCheckCapacity
+
 		additionalPr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "additional")
 		additionalPr.CreationTimestamp = metav1.NewTime(weekAgo)
 		additionalPr.Spec.ProvisioningClassName = v1beta1.ProvisioningClassCheckCapacity
-		processor := provReqProcessor{func() time.Time { return now }, 1, provreqclient.NewFakeProvisioningRequestClient(nil, t, pr, additionalPr), nil}
-		processor.refresh([]*provreqwrapper.ProvisioningRequest{pr, additionalPr})
+
+		oldFailedPr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "failed")
```
Comment (towca) [inline cluster-autoscaler/processors/provreq/processor_test.go:159]: IMO this should be a separate test case (which would require reworking this test a bit), or a separate test altogether. As it's right now it adds a lot of complexity to the test code for no apparent reason.

File: cluster-autoscaler/processors/provreq/processor_test.go [hunk 3]
```diff
@@ -172,6 +175,55 @@ func TestRefresh(t *testing.T) {
...
+		},
+	}
+	oldExpiredPr.Spec.ProvisioningClassName = v1.ProvisioningClassCheckCapacity
+
+	client := provreqclient.NewFakeProvisioningRequestClient(nil, t, pr, additionalPr, oldFailedPr, oldExpiredPr)
+
+	processor := provReqProcessor{func() time.Time { return now }, 1, client, nil}
+	processor.refresh([]*provreqwrapper.ProvisioningRequest{pr, additionalPr, oldFailedPr, oldExpiredPr})
+
+	_, err := client.ProvisioningRequestNoCache(oldFailedPr.Namespace, oldFailedPr.Name)
```
Comment (towca) [inline cluster-autoscaler/processors/provreq/processor_test.go:221]: nit: I'd also assert that the other 2 behave differently (or that these 2 behave differently before the refresh).

**Comment by towca** [comment]
/lgtm
/approve
/hold

Feel free to unhold if you don't want to address the nit.

File: cluster-autoscaler/processors/provreq/processor_test.go (hunk 3 above)
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/processors/provreq/processor_test.go:221]: Done

**Pull request merged**